cns/service/main.go

```diff
@@ -841,35 +843,48 @@ func main() {
 
 		logger.Printf("Set GlobalPodInfoScheme %v (InitializeFromCNI=%t)", cns.GlobalPodInfoScheme, cnsconfig.InitializeFromCNI)
 
-		err = InitializeCRDState(rootCtx, httpRestService, cnsconfig)
+		err = InitializeCRDState(rootCtx, httpRemoteRestService, cnsconfig)
 		if err != nil {
 			logger.Errorf("Failed to start CRD Controller, err:%v.\n", err)
 			return
 		}
 	}
 
 	// Initialize multi-tenant controller if the CNS is running in MultiTenantCRD mode.
-	// It must be started before we start HTTPRestService.
+	// It must be started before we start HTTPRemoteRestService.
 	if config.ChannelMode == cns.MultiTenantCRD {
-		err = InitializeMultiTenantController(rootCtx, httpRestService, *cnsconfig)
+		err = InitializeMultiTenantController(rootCtx, httpRemoteRestService, *cnsconfig)
 		if err != nil {
 			logger.Errorf("Failed to start multiTenantController, err:%v.\n", err)
 			return
 		}
+	}
 
+	logger.Printf("[Azure CNS] Start HTTP local server")
+	httpLocalRestService := restserverv2.New(httpRemoteRestService)
+	if httpLocalRestService != nil {
+		go func() {
+			err = httpLocalRestService.Start(rootCtx, defaultAPIServerURL)
+			if err != nil {
+				logger.Errorf("Failed to start echo server, err:%v.\n", err)
+				return
+			}
+		}()
 	}
 
-	logger.Printf("[Azure CNS] Start HTTP listener")
-	if httpRestService != nil {
+	logger.Printf("[Azure CNS] Start HTTP Remote server")
+	if httpRemoteRestService != nil {
 		if cnsconfig.EnablePprof {
-			httpRestService.RegisterPProfEndpoints()
+			httpRemoteRestService.RegisterPProfEndpoints()
 		}
 
-		err = httpRestService.Start(&config)
-		if err != nil {
-			logger.Errorf("Failed to start CNS, err:%v.\n", err)
-			return
-		}
+		go func() {
```

Comment:
we should not start a goroutine here since `httpRemoteRestService.Start` will start its own goroutine within